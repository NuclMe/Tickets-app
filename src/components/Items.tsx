import React from 'react';
import Box from '@mui/material/Box';
import { Item } from './Item';

export const Items = (props) => {
  const { items = [] } = props;
  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'column',
        gap: '20px',
        marginTop: '20px',
      }}
    >
      {items.map((i) => {
        return <Item key={i.price} {...i} />;
      })}
    </Box>
  );
};
