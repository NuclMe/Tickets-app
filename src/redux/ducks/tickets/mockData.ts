import type { ITicket } from '../../../types';

export const mockTickets: ITicket[] = [
  {
    price: 35230,
    carrier: 'SU',
    segments: [
      {
        origin: 'MOW',
        destination: 'HKT',
        date: '2018-05-18T02:00:00.000Z',
        stops: [],
        duration: 400,
      },
      {
        origin: 'HKT',
        destination: 'MOW',
        date: '2018-05-18T08:30:00.000Z',
        stops: [],
        duration: 300,
      },
    ],
  },
  {
    price: 30000,
    carrier: 'S7',
    segments: [
      {
        origin: 'MOW',
        destination: 'HKT',
        date: '2018-05-18T02:00:00.000Z',
        stops: ['LED'],
        duration: 600,
      },
      {
        origin: 'HKT',
        destination: 'MOW',
        date: '2018-05-18T08:30:00.000Z',
        stops: ['LED'],
        duration: 200,
      },
    ],
  },
  {
    price: 25000,
    carrier: 'TK',
    segments: [
      {
        origin: 'MOW',
        destination: 'HKT',
        date: '2018-05-18T02:00:00.000Z',
        stops: ['IST', 'DXB'],
        duration: 750,
      },
      {
        origin: 'HKT',
        destination: 'MOW',
        date: '2018-05-18T08:30:00.000Z',
        stops: ['IST', 'DXB'],
        duration: 650,
      },
    ],
  },
  {
    price: 40000,
    carrier: 'BA',
    segments: [
      {
        origin: 'MOW',
        destination: 'HKT',
        date: '2018-05-18T02:00:00.000Z',
        stops: ['LED'],
        duration: 500,
      },
      {
        origin: 'HKT',
        destination: 'MOW',
        date: '2018-05-18T08:30:00.000Z',
        stops: ['LED', 'AMS'],
        duration: 450,
      },
    ],
  },
  {
    price: 45000,
    carrier: 'LH',
    segments: [
      {
        origin: 'MOW',
        destination: 'HKT',
        date: '2018-05-18T02:00:00.000Z',
        stops: ['FRA', 'BKK', 'SIN'],
        duration: 800,
      },
      {
        origin: 'HKT',
        destination: 'MOW',
        date: '2018-05-18T08:30:00.000Z',
        stops: ['SIN', 'BKK', 'FRA'],
        duration: 750,
      },
    ],
  },
];
