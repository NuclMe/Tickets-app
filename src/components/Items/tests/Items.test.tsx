import { render } from '@testing-library/react';

import { Items } from '../Items';

import { mockData } from './mockData';

describe('Items', () => {
  it('renders the Items component via snapshot', () => {
    render(<Items items={mockData.tickets} />);

    expect(Items).toMatchSnapshot();
  });
});
