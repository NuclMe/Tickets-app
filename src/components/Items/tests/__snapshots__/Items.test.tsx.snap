// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Items > renders the Items component via snapshot 1`] = `[Function]`;
